yaptide/routes/common_sim_routes.py

```diff
@@ -168,28 +190,22 @@ def post():
         if decoded_token != sim_id:
             return yaptide_response(message="Invalid update key", code=400)
 
-        for estimator_dict in payload_dict["estimators"]:
-            # We forsee the possibility of the estimator being created earlier as element of partial results
-            estimator = fetch_estimator_by_sim_id_and_est_name(sim_id=sim_id, est_name=estimator_dict["name"])
-
-            if not estimator:
-                estimator = EstimatorModel(name=estimator_dict["name"], simulation_id=simulation.id)
-                estimator.data = estimator_dict["metadata"]
-                add_object_to_db(estimator)
-
-            for page_dict in estimator_dict["pages"]:
-                page = fetch_page_by_est_id_and_page_number(est_id=estimator.id,
-                                                            page_number=int(page_dict["metadata"]["page_number"]))
-
-                page_existed = bool(page)
-                if not page_existed:
-                    # create new page
-                    page = PageModel(page_number=int(page_dict["metadata"]["page_number"]), estimator_id=estimator.id)
-                # we always update the data
-                page.data = page_dict
-                if not page_existed:
-                    # if page was created, we add it to the session
-                    add_object_to_db(page, False)
+        if simulation.input_type == InputType.EDITOR.value:
+            outputs = simulation.inputs[0].data["input_json"]["scoringManager"]["outputs"]
+            sorted_estimator_names = sorted([output["name"] for output in outputs])
+            for output in outputs:
+                name = output["name"]
+                # estimator_dict is sorted alphabeticaly by names,
+                # thats why we can match indexes from sorted_estimator_names
+                estimator_dict_index = sorted_estimator_names.index(name)
+                estimator_dict = payload_dict["estimators"][estimator_dict_index]
+                create_or_update_estimator(sim_id=simulation.id, name=name, estimator_dict=estimator_dict)
```

Comment:
That loop makes as many commits as we have outputs. If user defines hundred of output files, then we'll have hundred of commits.

Do you think this may be optimised by first running a loop over all outputs and do `add_object_to_db(estimator, make_commit=False)`

once that is done, make a single commit to the DB

Then again make a loop over all outputs and handle pages. I think that needs to query DB, therefore commit made above is needed.

This way you could have logic in `prepare_create_or_update_estimator_in_db` and `prepare_create_or_update_pages_in_db`. I'd also use some better name for `create_or_update_estimator` (i.e. `_in_db` suffix) to make clear such function is not modifying the DB (which is expensive), but prepares a ground to make a commit.